File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxsum = current_sum = nums[0]


        for num in nums[1:]:

            current_sum= max(num,current_sum+num)

            maxsum = max(current_sum,maxsum)

        return maxsum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        int current_sum =nums[0];
        int max_sum =nums[0];

        for(int i=1;i<=nums.length-1;i++)
        {
            current_sum=Math.max(nums[i],nums[i]+current_sum); //decision to select a current number or not 1. we will selection only when if the current sum is greater than the previous one. 
            max_sum=Math.max(max_sum,current_sum);
        }

      return max_sum;


    }
}